staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go

```diff
@@ -142,11 +152,40 @@ func (c *DiscoveryController) sync(version schema.GroupVersion) error {
 			StorageVersionHash: storageVersionHash,
 		})
 
+		var scope apidiscoveryv2beta1.ResourceScope
+		if crd.Spec.Scope == apiextensionsv1.NamespaceScoped {
+			scope = apidiscoveryv2beta1.ScopeNamespace
+		} else {
+			scope = apidiscoveryv2beta1.ScopeCluster
+		}
+		apiResourceDiscovery = apidiscoveryv2beta1.APIResourceDiscovery{
```

Comment:
When we enter the loop for the second time, this gets overwitten?  How does this work when we have multiple CRDs that need to be added for a single groupversion?  Say foos and bars are both in v1 of cool-group.com

Reply:
Fixed and added a test case